service/history/visibility_queue_task_executor.go

```diff
@@ -496,6 +336,41 @@ func (t *visibilityQueueTaskExecutor) processDeleteExecution(
 	return t.visibilityMgr.DeleteWorkflowExecution(ctx, request)
 }
 
+// getVisibilityRequestBase builds the VisibilityRequestBase from the mutable state.
+// This function must do a deep copy of the values extracted from the mutable state since
+// to ensure that the values won't be changed by accident.
```

Comment:
```suggestion
// getVisibilityRequestBase builds the VisibilityRequestBase from the mutable state.
// This function must do a deep copy of the values extracted from the mutable state
// to make sure that mutable state is not accessed after workflow lock is released and // there is no data race.
```

And move it to line 354. No one reads function comments.